bin/node-template/node/src/service.rs

```diff
@@ -87,9 +89,11 @@ pub fn new_full(config: Configuration<GenesisConfig>)
 			.expect("Link Half and Block Import are present for Full Services or setup failed before. qed");
 
 	let service = builder
-		.with_finality_proof_provider(|client, backend|
-			Ok(Arc::new(GrandpaFinalityProofProvider::new(backend, client)) as _)
-		)?
+		.with_finality_proof_provider(|client, backend| {
+			// GenesisAuthoritySetProvider is implemented for StorageAndProofProvider
+			let provider = client as Arc<dyn StorageAndProofProvider<_, _>>;
+			Ok(Arc::new(GrandpaFinalityProofProvider::new(backend, provider)) as _)
```

Comment:
```suggestion
			Ok(Arc::new(GrandpaFinalityProofProvider::new(backend, client.clone() as _)) as _)
```
This should probably work fine as well (without the upper line).

Reply:
the compiler cannot infer the type, needs the cast to `StorageAndProofProvider` -> `GenesisAuthoritySetProvider.`